app/services/dwp/benefit_check_status_service.rb

```diff
@@ -22,24 +22,28 @@ def call
     attr_reader :crime_application, :person
 
     def benefit_check_status
-      return BenefitCheckStatus::NO_CHECK_NO_NINO.to_s if nino_forthcoming?
-      return undetermined_status if dwp_undetermined
       return BenefitCheckStatus::NO_CHECK_REQUIRED.to_s if benefit_check_subject.benefit_type == 'none'
+      return BenefitCheckStatus::NO_CHECK_NO_NINO.to_s if nino_forthcoming?
       return BenefitCheckStatus::CHECKER_UNAVAILABLE.to_s if checker_down
+      return BenefitCheckStatus::UNDETERMINED.to_s if undetermined
+      return BenefitCheckStatus::NO_RECORD_FOUND.to_s if not_in_receipt
 
-      BenefitCheckStatus::CONFIRMED.to_s if benefit_check_subject.benefit_check_result
+      BenefitCheckStatus::CONFIRMED.to_s if confirmed
     end
 
-    def undetermined_status
-      return BenefitCheckStatus::UNDETERMINED.to_s if benefit_evidence_forthcoming?
+    def undetermined
+      benefit_check_subject.dwp_response == 'Undetermined' || (benefit_check_subject.confirm_dwp_result == 'no' && benefit_evidence_forthcoming?)
+    end
 
-      BenefitCheckStatus::NO_RECORD_FOUND.to_s if means_assessment_as_benefit_evidence?
+    def not_in_receipt
+      benefit_check_subject.dwp_response == 'No' || (benefit_check_subject.confirm_dwp_result == 'no' && means_assessment_as_benefit_evidence?)
     end
```

Comment:
Would it be better have a method for `dwp_response` on the model that would calculate this? My only concern would be that if the app was returned it would be submitted with the calculated attribute `dwp_response` when that was not actually generated from the checker 🤔